### PR TITLE
Dtspb 1526

### PR DESCRIPTION
**Pull request by nigeldunne**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/DTSPB-1526

### Change description ###
DTSPB-1526: Life Event Verification

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[X] No
```

### CODE REVIEW

**Review by isha2040 — APPROVED** [review]

File: src/functionalTest/java/uk/gov/hmcts/probate/functional/lifeeventservice/LifeEventServiceTest.java [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+import io.restassured.response.Response;
+import net.serenitybdd.junit.spring.integration.SpringIntegrationSerenityRunner;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import uk.gov.hmcts.probate.functional.IntegrationTestBase;
+
+import static junit.framework.TestCase.assertEquals;
+
+@RunWith(SpringIntegrationSerenityRunner.class)
+public class LifeEventServiceTest extends IntegrationTestBase {
```
Comment (isha2040) [inline src/functionalTest/java/uk/gov/hmcts/probate/functional/lifeeventservice/LifeEventServiceTest.java:13]: should we not have any more FTs for the fail scenario too ?

File: src/main/java/uk/gov/hmcts/probate/service/LifeEventService.java [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+        log.info("Trying LEV call");
+        List<V1Death> records = emptyList();
+        try {
+            records = deathService
+                    .searchForDeathRecordsByNamesAndDate(deceasedForenames, deceasedSurname, deceasedDateOfDeath);
+        } catch (Exception e) {
+            log.error("Error during LEV call", e);
+        }
+        log.info("Records returned: " + records.size());
+        if (1 == records.size()) {
```
Comment (isha2040) [inline src/main/java/uk/gov/hmcts/probate/service/LifeEventService.java:59]: should we not explicitly handle the scenario where there are more than one record returned under a else condition

File: src/main/java/uk/gov/hmcts/probate/service/ccd/CcdClientApi.java [hunk 3]
```diff
@@ -87,10 +89,43 @@ public CaseDetails updateCaseAsCaseworker(CcdCaseType caseType, String caseId, C
...
+                caseDataContent
+        );
+    }
+
+    @Override
+    public CaseDetails updateCaseAsCitizen(CcdCaseType caseType, String caseId, CaseData caseData, EventId eventId,
+                                              SecurityDTO securityDTO, String description, String summary) {
+        log.info("Update case as citizen for caseType: {}, caseId: {}, eventId: {}",
+                caseType.getName(), caseId, eventId.getName());
+        log.info("Retrieve event token from CCD for citizen, caseType: {}, caseId: {}, eventId: {}",
```
Comment (isha2040) [inline src/main/java/uk/gov/hmcts/probate/service/ccd/CcdClientApi.java:111]: can we add lev keyword too in this message to make it easy to associate to lev calls

File: src/main/java/uk/gov/hmcts/probate/service/LifeEventService.java (hunk 2 above)
Comment (nigeldunne) [reply src/main/java/uk/gov/hmcts/probate/service/LifeEventService.java:59]: We will do in the next release under jira https://tools.hmcts.net/jira/browse/DTSPB-1577

File: src/main/java/uk/gov/hmcts/probate/service/ccd/CcdClientApi.java (hunk 3 above)
Comment (nigeldunne) [reply src/main/java/uk/gov/hmcts/probate/service/ccd/CcdClientApi.java:111]: Added an additional log message to LifeEventService. 
This method already logs the eventId:

2021-May-12 13:21:59.819 INFO  [task-3] u.g.h.p.s.c.CcdClientApi - Update case as citizen for caseType: GrantOfRepresentation, caseId: 1620825628357285, eventId: deathRecordVerified

Comment (nigeldunne) [reply src/main/java/uk/gov/hmcts/probate/service/ccd/CcdClientApi.java:111]: Removed redundant log info

**Review by isha2040 — APPROVED** [review]

**Pull request merged**